scripts/confix/confix.go

```diff
@@ -0,0 +1,214 @@
+// Program confix applies fixes to a Tendermint TOML configuration file to
+// update a file created with an older version of Tendermint to a compatible
+// format for a newer version.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/creachadair/atomicfile"
+	"github.com/creachadair/tomledit"
+	"github.com/creachadair/tomledit/parser"
+	"github.com/creachadair/tomledit/transform"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %[1]s [options] -config <path>
+
+Modify the contents of the specified -config TOML file to update the
+names, locations, and values of configuration settings to the current
+configuration layout.
+
+By default, the config file is edited in-place; use -out to write the
+modified file to a different path. In case of any error in updating the
+file, the input is not modified.
+
+Options:
+`, filepath.Base(os.Args[0]))
+		flag.PrintDefaults()
+	}
+}
+
+var (
+	configPath = flag.String("config", "", "Config file path (required)")
+	outPath    = flag.String("out", "", "Output file path (defaults to -input)")
+)
+
+func main() {
+	flag.Parse()
+	if *configPath == "" {
+		log.Fatal("You must specify a non-empty -config path")
+	} else if *outPath == "" {
+		*outPath = *configPath
+	}
+
+	doc, err := LoadConfig(*configPath)
+	if err != nil {
+		log.Fatalf("Loading config: %v", err)
+	}
+
+	ctx := transform.WithLogWriter(context.Background(), os.Stderr)
+	if err := ApplyFixes(ctx, doc); err != nil {
+		log.Fatalf("Updating %q: %v", *configPath, err)
+	}
+
+	out, err := atomicfile.New(*outPath, 0600)
+	if err != nil {
+		log.Fatalf("Open output: %v", err)
+	}
+	defer out.Cancel()
+
+	if err := tomledit.Format(out, doc); err != nil {
```

Comment:
I think we should verify that Tendermint correctly parses and reads the config here, no?

Reply:
That's an excellent idea

Reply:
This is done. I had hoped to call the same function the node itself uses, but that turns out to be impractical because of all the implicit defaults in viper. But I think I got a faithful-enough emulation for it to be valuable (in particular, it uses the real config type and validation logic).